Yocto 5 Upgrade

UtsavAgarwalADI opened the pull request:
Added Compatibility with Yocto 5.0 (Scarthgap)
Added Compatibility with upstream (adsp-main)

In order to build with upstream add the following to your local.conf
``` 
ADSP_KERNEL_TYPE="upstream" 
```

NOTE: Due to limited upstream support, this new version of the kernel can only be built for SC598

mattsm commented:
@UtsavAgarwalADI can we break these into more commits? I think it will help to document things one-by-one.

UtsavAgarwalADI commented:
> @UtsavAgarwalADI can we break these into more commits? I think it will help to document things one-by-one.

I'll aim to break them by subsystem and force push to the same branch

Utsav-Agarwal committed:
Layer compatibility

Changing meta layer compatibility from yocto 4(kirkstone) to
yocto 5(scarthgap)

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
rootfs: Updating rootfs file names

yocto 5 seems to append .rootfs. onto previously generated
filesystem names:
	- Changing initramfs name in fitImage's .its
	- Changing expected rootfs name in wic script

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
ADI Crypto: Adding patch upstream status

Adding upstream status for crypto subsystem patches based on yocto 5 QA requirements

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
testptp: Refreshing patch

Refreshing patch for testptp based on yocto 5 head

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
busybox: Updating defconfig and watchdog

Disabling CONFIG_SETARCH since it leads to installation postscript
error

Removing excess bracket which could have led to an error/unexpected
behaviour

Renaming systemd recipe based on version

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
python: Removing unneeded patch

Removing patch disabling NIS module

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
rng-tools: refreshing patch

Refreshing rng-tools patch for disabling RNDR based on yocto-5 head

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
ICC: Updating firmware location

Updating expected firmware location based on yocto-5

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
Linux kernel: Updating firmware location

Updating expected firmware location based on yocto-5

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
alsa-lib: Refreshing and adding patch status

Refreshing pcm patch based on yocto-5 head

Adding upstream status for patches based on yocto-5 QA requirements

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
Updating cryptodev

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
Linux kernel: Adding upstream kernel support

- Updating SRCREV for cryptodev as needed
	- Upstream kernel can be built by adding
		ADSP_KERNEL_TYPE="upstream"

NOTE: Upstream kernel is only applicable for sc598 boards

NOTE: Currently perf reports a compilation error with kenrel 6.9
As a result, it is disabled for now to allow full image
compilation.

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
liburcu: removing patch to disable test

Signed-off-by: UtsavAgarwalADI <utsav.agarwal@analog.com>

Utsav-Agarwal committed:
cryptodev: fixing cryptodev commit id

Fixing cryptodev's commit id for compatibility with "upstream" kernel

Signed-off-by: Utsav Agarwal <utsav.agarwal@analog.com>

jdanielllopis commented:
reopen later
